src/stdlib_str2num.f90

```diff
@@ -0,0 +1,532 @@
+        !----------------------------------------------
+        call to_num_base(s,r,p,err)
+        p = min( p , len(s) )
+        s => s(p:)
+        if(present(stat)) stat = err
+    end function
+
+    elemental function to_double(s,mold) result(r)
+        ! -- In/out Variables
+        character(*), intent(in) :: s !> input string
+        real(dp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(dp) :: r    !> Output real value
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: stat ! error status
+        !----------------------------------------------
+        call to_num_base(s,r,p,stat)
+    end function
+    
+    function to_double_p(s,mold,stat) result(r)
+        ! -- In/out Variables
+        character(len=:), pointer :: s !> input string
+        real(dp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(dp) :: r    !> Output real value
+        integer(1),intent(inout), optional :: stat
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: err
+        !----------------------------------------------
+        call to_num_base(s,r,p,err)
+        p = min( p , len(s) )
+        s => s(p:)
+        if(present(stat)) stat = err
+    end function
+
+    function to_quad(s,mold) result(r)
+        ! -- In/out Variables
+        character(*), intent(in) :: s !> input string
+        real(qp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(qp) :: r    !> Output real value
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: stat ! error status
+        !----------------------------------------------
+        call to_num_base(s,r,p,stat)
+    end function
+    
+    function to_quad_p(s,mold,stat) result(r)
+        ! -- In/out Variables
+        character(len=:), pointer :: s !> input string
+        real(qp), intent(in) :: mold !> dummy argument to disambiguate at compile time the generic interface
+        real(qp) :: r    !> Output real value
+        integer(1),intent(inout), optional :: stat
+        ! -- Internal Variables
+        integer(1) :: p !> position within the number
+        integer(1) :: err
+        !----------------------------------------------
+        call to_num_base(s,r,p,err)
+        p = min( p , len(s) )
+        s => s(p:)
+        if(present(stat)) stat = err
+    end function
+
+    !---------------------------------------------
+    ! String To Number Implementations
+    !---------------------------------------------
+
+    elemental subroutine to_int_32(s,v,p,stat)
+        !> Return an unsigned 32-bit integer
+        ! -- In/out Variables
+        character(*), intent(in) :: s !> input string
+        integer, intent(inout)  :: v !> Output real value
```

Comment:
```suggestion
        integer(int32), intent(inout)  :: v !> Output real value
```

Would the same algorithm be used for e.g., `int8` or `int64`? If yes, then it could be easily extended with `fypp`.

Reply:
Yes it could, we could indeed extended with fypp